substrate/frame/support/src/lib.rs

```diff
@@ -2262,6 +2262,221 @@ pub mod pallet_macros {
 	/// }
 	/// ```
 	pub use frame_support_procedural::genesis_build;
+
+	/// The `#[pallet::constant]` attribute can be used to add an associated type trait bounded by `Get`
+	/// from [`pallet::config`](`macro@config`) into metadata.
+	///
+	/// ## Example:
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     use frame_support::pallet_prelude::*;
+	///
+	///     # #[pallet::pallet]
+	///     # pub struct Pallet<T>(_);
+	///
+	///     #[pallet::config]
+	///     pub trait Config: frame_system::Config {
+	/// 		/// This is like a normal `Get` trait, but it will be added into metadata.
+	/// 		#[pallet::constant]
+	/// 		type Foo: Get<u32>;
+	/// 	}
+	/// }
+	/// ```
+	pub use frame_support_procedural::constant;
+
+	// TODO: Add documentation
+	pub use frame_support_procedural::constant_name;
+
+	/// Declares an implementation block to be the callable portion of a pallet. In other words,
+	/// each function in the implementation block of this section is an extrinsic that can be called
+	/// externally.
+	///
+	/// Other than the `fn` attached to `Pallet`, this block will also generate an `enum Call`
+	/// which encapsulates the different functions, alongside their arguments, except for `origin`.
+	/// [`sp_runtime::traits::Dispatchable`] is then implemented for `enum Call`.
+	///
+	/// ## Example:
+	///
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	///     use frame_support::pallet_prelude::*;
+	///
+	///     #[pallet::config]
+	///     pub trait Config: frame_system::Config { }
+	///
+	///     #[pallet::pallet]
+	///     pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::call]
+	/// 	impl<T: Config> Pallet<T> {
+	/// 		fn do_stuff(_origin: OriginFor<T>, arg: u32) {
+	/// 			unimplemented!()
+	/// 		}
+	/// 	}
+	/// }
+	/// ```
+	///
+	/// TODO: Once we have default pallet config, it would also be easy to create types in the example
+	/// that implement `Config`.
+	pub use frame_support_procedural::call;
+
+	/// Declares a type alias as a storage item. Storage items are pointers to data stored on-chain (the
+	/// *blockchain state*), under a specific key. The exact key is dependent on the type of the
```

Comment:
Yes I want to double down that the key location should be documented *somewhere* in this story.